virtual_rainforest/core/grid.py

```diff
@@ -1,203 +1,370 @@
 """The `core.grid` module.
 
 The `core.grid` module is used to create the grid of cells underlying the simulation and
-to set up the neighbourhood connections of cells.
+to identify the neighbourhood connections of cells.
 
 TODO - set up neighbourhoods. ? store as graph (networkx - might only need a really
        lightweight graph description).
-TODO - export of files to geojson
-TODO - import of geojson grids? Way to link structured landscape into cells
+TODO - import of geojson grids? Way to link structured landscape into cells.  Can use
+       data loading methods to assign values to grids? This would be a useful way of
+       defining mappings though.
 """
 
-import fiona
+import json
+import logging
+from functools import wraps
+from typing import Callable
+
 import numpy as np
-from pydantic import StrictInt
-from pydantic.dataclasses import dataclass
-from pydantic.typing import Literal
+from shapely.affinity import scale, translate
+from shapely.geometry import Polygon
 
+LOGGER = logging.getLogger("virtual_rainforest.core")
 
-@dataclass
-class CoreGridConfig:
-    """Configure the `core.grid` module.
+GRID_REGISTRY: dict[str, Callable] = {}
+"""A registry for different grid geometries.
 
-    This data class is used to setup the arrangement of grid cells to be used in
-    running a `virtual_rainforest` simulation.
+This dictionary maps grid geometry types (square, hex, etc) to a function generating a
+grid of that type. Users can register their own grid types using the `register_grid`
+decorator.
+"""
 
-    Attrs:
-        grid_type:
-        cell_contiguity:
-        cell_area:
-        cell_nx:
-        cell_ny:
-    """
 
-    grid_type: Literal["hex", "square"] = "square"
-    cell_contiguity: Literal["queen", "rook"] = "rook"
-    cell_area: float = 100
-    cell_nx: StrictInt = 10
-    cell_ny: StrictInt = 10
+def register_grid(grid_type: str):
```

Comment:
I think you need to be explicit here about the return type of this decorator and the inner one so the signature expected for the decorated function is clear.